tests/tables/test_SlowlyChangingDimension_lookupasof.py

```diff
@@ -0,0 +1,104 @@
+import unittest
+
+from tests import utilities
+import pygrametl
+import pygrametl.drawntabletesting as dtt
+from pygrametl.tables import SlowlyChangingDimension
+
+
+class SlowlyChangingDimensionLookupasofTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        utilities.ensure_default_connection_wrapper()
+        cls.initial = dtt.Table("customers", """
```

Comment:
The test data should be extended to include all cases, e.g., `fromdate` `NULL` does not seem to be included. A tool like [`coverage.py`](https://github.com/nedbat/coveragepy) can be used to check that all of the new methods have been checked.

Reply:
Yes, the tests are currently not exhaustive

Reply:
More tests added